Implement torus volume mesh

tjhei opened the pull request.

luca-heltai commented on include/deal.II/grid/grid_generator.h:
```diff
@@ -761,9 +761,18 @@ namespace GridGenerator
-   * of this object can be described by the TorusBoundary class.
+   * Produce the volume or surface mesh of a torus. The axis of the torus is
+   * the $y$-axis while the plane of the torus is the $x$-$z$ plane. The
+   * function is implemented
```
Missing end of phrase?

luca-heltai commented on tests/grid/torus_01.cc:
```diff
@@ -0,0 +1,91 @@
+  std::ofstream out ("grid-1.vtk");
+  GridOut grid_out;
+  grid_out.write_vtk (triangulation, out);
+  */
```
Remove if you don't need it.

luca-heltai commented on tests/grid/torus_01.cc:
```diff
@@ -0,0 +1,91 @@
+  triangulation.set_manifold (0, desc_torus);
+  triangulation.set_manifold (1, desc_sphere);
+
+  triangulation.refine_global(0);
```
0?

tjhei commented:
Addressed your comments.

@luca-heltai  Is it okay to advertise TorusManifold for a <2,3> torus instead of TorusBoundary? Or should I suggest TorusBoundary for now?

luca-heltai commented:
I think we can start to advertise `TorusManifold<2,3>` instead of `TorusBoundary<2,3>`

bangerth commented:
Would you mind adding an entry to changes.h in a separate PR?
